lib/features/thread/presentation/thread_view.dart

```diff
@@ -654,17 +667,26 @@ class ThreadView extends GetWidget<ThreadController>
   Widget _buildEmptyEmail(BuildContext context) {
     return Obx(() => controller.viewState.value.fold(
       (failure) => const SizedBox.shrink(),
-      (success) => success is! LoadingState && success is! SearchingState
-        ? EmptyEmailsWidget(
-            key: const Key('empty_thread_view'),
-            title: _getMessageEmptyEmail(context),
-            iconSVG: controller.imagePaths.icEmptyEmail,
-            subTitle: _getSubMessageEmptyEmail(context),
-            onCreateFiltersActionCallback: controller.isNewFolderCreated
-              ? controller.goToCreateEmailRuleView
-              : null,
-          )
-        : const SizedBox.shrink())
+      (success) {
+        if (success is! GetAllEmailLoading && success is! SearchingState) {
+          if (success is GetAllEmailSuccess
+              && success.currentMailboxId != controller.selectedMailboxId) {
+            return const SizedBox.shrink();
+          } else {
+            return EmptyEmailsWidget(
+              key: const Key('empty_thread_view'),
+              title: _getMessageEmptyEmail(context),
+              iconSVG: controller.imagePaths.icEmptyEmail,
+              subTitle: _getSubMessageEmptyEmail(context),
+              onCreateFiltersActionCallback: controller.isNewFolderCreated
+                ? controller.goToCreateEmailRuleView
+                : null,
+            );
+          }
+        } else {
+          return const SizedBox.shrink();
+        }
```

Comment:
```suggestion
        if (success is GetAllEmailLoading || success is SearchingState) {
          return const SizedBox.shrink();
        }
        
        if (success is GetAllEmailSuccess
            && success.currentMailboxId != controller.selectedMailboxId) {
          return const SizedBox.shrink();
        } else {
          return EmptyEmailsWidget(
            key: const Key('empty_thread_view'),
            title: _getMessageEmptyEmail(context),
            iconSVG: controller.imagePaths.icEmptyEmail,
            subTitle: _getSubMessageEmptyEmail(context),
            onCreateFiltersActionCallback: controller.isNewFolderCreated
              ? controller.goToCreateEmailRuleView
              : null,
          );
        }
```

Avoid nesting `if` statements

Reply:
done